src/d2q9_cumulant/Dynamics.c

```diff
@@ -155,50 +156,51 @@ CudaDeviceFunction void EPressure()
 CudaDeviceFunction void CollisionMRT()
 {
 
- real_t c[9],m[9];
-  real_t u[2],usq,d;
-  d = getRho();
-  u[0] = (f[8] - f[7] - f[6] + f[5] - f[3] + f[1]);
-  u[1] = (-f[8] - f[7] + f[6] + f[5] - f[4] + f[2]);
-  usq = u[0]*u[0] + u[1]*u[1];
-   real_t  w[5] = {1.0/(3*nu+0.5),1.,1.,1.,1.0};  // defining relaxation rate for first cummulants
+ real_t c[9],d;
+ d = getRho();
+ real_t  w[5] = {1.0/(3*nu+0.5),1.,1.,1.,1.0};  // defining relaxation rate for first cummulants
  if ((NodeType & NODE_BOUNDARY) != 0) w[0] = 1.0/(3*nubuffer+0.5);
 
-for (int i = 0;i<9;i++) m[i] = f[i];
 //First determing moments from density-probability function
   
-  m[0] = m[3] + m[1] + m[0]; 
-  m[1] = -m[3] + m[1];
-  m[3] = m[1] + m[3]*2.; 
-  m[2] = m[6] + m[5] + m[2];
-  m[5] = -m[6] + m[5];
-  m[6] = m[5] + m[6]*2.; 
-  m[4] = m[7] + m[8] + m[4];
-  m[8] = -m[7] + m[8];
-  m[7] = m[8] + m[7]*2.; 
-  m[0] = m[4] + m[2] + m[0];
-  m[2] = -m[4] + m[2];
-  m[4] = m[2] + m[4]*2.; 
-  m[1] = m[8] + m[5] + m[1];
-  m[5] = -m[8] + m[5];
-  m[8] = m[5] + m[8]*2.; 
-  m[3] = m[7] + m[6] + m[3];
-  m[6] = -m[7] + m[6];
-  m[7] = m[6] + m[7]*2.; 
+  f[0] = f[3] + f[1] + f[0]; 
+  f[1] = -f[3] + f[1];
+  f[3] = f[1] + f[3]*2.; 
+  f[2] = f[6] + f[5] + f[2];
+  f[5] = -f[6] + f[5];
+  f[6] = f[5] + f[6]*2.; 
+  f[4] = f[7] + f[8] + f[4];
+  f[8] = -f[7] + f[8];
+  f[7] = f[8] + f[7]*2.; 
+  f[0] = f[4] + f[2] + f[0];
+  f[2] = -f[4] + f[2];
+  f[4] = f[2] + f[4]*2.; 
+  f[1] = f[8] + f[5] + f[1];
+  f[5] = -f[8] + f[5];
+  f[8] = f[5] + f[8]*2.; 
+  f[3] = f[7] + f[6] + f[3];
+  f[6] = -f[7] + f[6];
+  f[7] = f[6] + f[7]*2.; 
   
-//Cummulant calculation from moments
-  c[0] = m[0];
-  c[1] = m[1]/m[0];
-  c[3] = ( -c[1]*m[1] + m[3] )/m[0];
-  c[2] = m[2]/m[0];
-  c[5] = ( -c[1]*m[2] + m[5] )/m[0];
-  c[6] = ( -c[5]*m[1] - c[3]*m[2] - c[1]*m[5] + m[6] )/m[0];
-  c[4] = ( -c[2]*m[2] + m[4] )/m[0];
-  c[8] = ( -c[1]*m[4] + m[8] - c[5]*m[2]*2. )/m[0];
-  c[7] = ( -c[8]*m[1] - c[3]*m[4] - c[1]*m[8] + m[7] + ( -c[6]*m[2] - c[5]*m[5] )*2. )/m[0];
+//Cuffulant calculation frof fofents
```

Comment:
s/m/f/g :D

2016-06-24 15:28 GMT+02:00 Łukasz Łaniewski-Wołłk <notifications@github.com>
:

> In src/d2q9_cumulant/Dynamics.c
> <https://github.com/CFD-GO/TCLB/pull/73#discussion_r68397431>:
>
> >
> > -//Cummulant calculation from moments
> > -  c[0] = m[0];
> > -  c[1] = m[1]/m[0];
> > -  c[3] = ( -c[1]*m[1] + m[3] )/m[0];
> > -  c[2] = m[2]/m[0];
> > -  c[5] = ( -c[1]*m[2] + m[5] )/m[0];
> > -  c[6] = ( -c[5]*m[1] - c[3]*m[2] - c[1]*m[5] + m[6] )/m[0];
> > -  c[4] = ( -c[2]*m[2] + m[4] )/m[0];
> > -  c[8] = ( -c[1]*m[4] + m[8] - c[5]*m[2]*2. )/m[0];
> > -  c[7] = ( -c[8]*m[1] - c[3]*m[4] - c[1]*m[8] + m[7] + ( -c[6]*m[2] - c[5]*m[5] )*2. )/m[0];
> > +//Cuffulant calculation frof fofents
>
> Cuffulants frof fofents? :D
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/CFD-GO/TCLB/pull/73/files/25b0f47deb603f9ab1d0f24ce7caaae622df40a2#r68397431>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ADYOxydcWEkJHhJU6HyJFvZgdtLmtMwKks5qO9uUgaJpZM4I9hjQ>
> .
>